### ISSUE TITLE
ONNX model converted from a TensorFlow model (with saved_model format) isn't read by readNetFromONNX

### ISSUE THREAD

**Issue by JJonahJson** [opencv/opencv#19366]
##### System information (version)

- OpenCV => 4.5.1.48
- Operating System / Platform => Google Colab

##### Detailed description

I've got the following error:
 OpenCV(4.5.1) /tmp/pip-req-build-1syr35c1/opencv/modules/dnn/src/onnx/onnx_graph_simplifier.cpp:592: error: (-210:Unsupported format or combination of formats) Unsupported data type: BOOL in function 'getMatFromTensor'

because in onnx_graph_simplifier.cpp the TensorProto_DataType_BOOL isn't mentioned, will it be added soon?

The model I converted is Mask R-CNN Inception Resnet V2, you can download it from this link:

http://download.tensorflow.org/models/object_detection/tf2/20200711/mask_rcnn_inception_resnet_v2_1024x1024_coco17_gpu-8.tar.gz

I trained it with TensorFlow 2, I've used the saved_model format and converted it to ONNX doing the following:

##### Steps to reproduce
```pip install tf2onnx
pip install opencv-python==4.5.1.48
python -m tf2onnx.convert --saved-model /path/to/saved_model/dir/ --opset 13 --output /path/to/model.onnx


import cv2
net = cv2.dnn.readNetFromONNX('/path/to/model.onnx')   #This is where the error raises
image = cv2.imread("/path/to/image_example.jpg")
blob = cv2.dnn.blobFromImage(image)
net.setInput(blob)
net.forward()
```

##### Issue submission checklist

 - [X] I report the issue, it's not a question
   
 - [X] I checked the problem with documentation, FAQ, open issues,
       forum.opencv.org, Stack Overflow, etc and have not found solution
   
 - [X] I updated to latest OpenCV version and the issue is still there
   
 - [X] There is reproducer code and related data files: videos, images, onnx, etc

**Comment by Byronnar** [comment]
Hi, Have you solved this problem? 

> ##### System information (version)
> * OpenCV => 4.5.1.48
> * Operating System / Platform => Google Colab
> 
> ##### Detailed description
> I've got the following error:
> OpenCV(4.5.1) /tmp/pip-req-build-1syr35c1/opencv/modules/dnn/src/onnx/onnx_graph_simplifier.cpp:592: error: (-210:Unsupported format or combination of formats) Unsupported data type: BOOL in function 'getMatFromTensor'
> 
> because in onnx_graph_simplifier.cpp the TensorProto_DataType_BOOL isn't mentioned, will it be added soon?
> 
> The model I converted is Mask R-CNN Inception Resnet V2, you can download it from this link:
> 
> http://download.tensorflow.org/models/object_detection/tf2/20200711/mask_rcnn_inception_resnet_v2_1024x1024_coco17_gpu-8.tar.gz
> 
> I trained it with TensorFlow 2, I've used the saved_model format and converted it to ONNX doing the following:
> 
> ##### Steps to reproduce
> ```
> pip install tf2onnx
> pip install opencv-python==4.5.1.48
> python -m tf2onnx.convert --saved-model /path/to/saved_model/dir/ --opset 13 --output /path/to/model.onnx
> 
> 
> import cv2
> net = cv2.dnn.readNetFromONNX('/path/to/model.onnx')   #This is where the error raises
> image = cv2.imread("/path/to/image_example.jpg")
> blob = cv2.dnn.blobFromImage(image)
> net.setInput(blob)
> net.forward()
> ```
> 
> ##### Issue submission checklist
> * [x]  I report the issue, it's not a question
> * [x]  I checked the problem with documentation, FAQ, open issues,
>   forum.opencv.org, Stack Overflow, etc and have not found solution
> * [x]  I updated to latest OpenCV version and the issue is still there
> * [x]  There is reproducer code and related data files: videos, images, onnx, etc

My error are below:
```
 net = cv2.dnn.readNetFromONNX('./mask_rcnn.onnx')   #This is where the error raises
cv2.error: OpenCV(4.5.1) /tmp/pip-req-build-1syr35c1/opencv/modules/dnn/src/onnx/onnx_graph_simplifier.cpp:592: error: (-210:Unsupported format or combination of formats) Unsupported data type: BOOL in function 'getMatFromTensor'

```

**Comment by JJonahJson** [comment]
> Hi, Have you solved this problem?
> 
> > ##### System information (version)
> > 
> > * OpenCV => 4.5.1.48
> > * Operating System / Platform => Google Colab
> > 
> > ##### Detailed description
> > I've got the following error:
> > OpenCV(4.5.1) /tmp/pip-req-build-1syr35c1/opencv/modules/dnn/src/onnx/onnx_graph_simplifier.cpp:592: error: (-210:Unsupported format or combination of formats) Unsupported data type: BOOL in function 'getMatFromTensor'
> > because in onnx_graph_simplifier.cpp the TensorProto_DataType_BOOL isn't mentioned, will it be added soon?
> > The model I converted is Mask R-CNN Inception Resnet V2, you can download it from this link:
> > http://download.tensorflow.org/models/object_detection/tf2/20200711/mask_rcnn_inception_resnet_v2_1024x1024_coco17_gpu-8.tar.gz
> > I trained it with TensorFlow 2, I've used the saved_model format and converted it to ONNX doing the following:
> > ##### Steps to reproduce
> > ```
> > pip install tf2onnx
> > pip install opencv-python==4.5.1.48
> > python -m tf2onnx.convert --saved-model /path/to/saved_model/dir/ --opset 13 --output /path/to/model.onnx
> > 
> > 
> > import cv2
> > net = cv2.dnn.readNetFromONNX('/path/to/model.onnx')   #This is where the error raises
> > image = cv2.imread("/path/to/image_example.jpg")
> > blob = cv2.dnn.blobFromImage(image)
> > net.setInput(blob)
> > net.forward()
> > ```
> > 
> > 
> > ##### Issue submission checklist
> > 
> > * [x]   I report the issue, it's not a question
> > * [x]   I checked the problem with documentation, FAQ, open issues,
> >   forum.opencv.org, Stack Overflow, etc and have not found solution
> > * [x]   I updated to latest OpenCV version and the issue is still there
> > * [x]   There is reproducer code and related data files: videos, images, onnx, etc
> 
> My error are below:
> 
> ```
>  net = cv2.dnn.readNetFromONNX('./mask_rcnn.onnx')   #This is where the error raises
> cv2.error: OpenCV(4.5.1) /tmp/pip-req-build-1syr35c1/opencv/modules/dnn/src/onnx/onnx_graph_simplifier.cpp:592: error: (-210:Unsupported format or combination of formats) Unsupported data type: BOOL in function 'getMatFromTensor'
> ```

No, in the meantime I'm trying to train a TF1 model since it's well supported by OpenCV  unlike TF2. Sadly I cannot obtain the same result both speaking of mAP and inference speed.

**Comment by LupusSanctus** [comment]
@JJonahJson, could you, please, attach your `.pb` model?

**Comment by JJonahJson** [comment]
> @JJonahJson, could you, please, attach your `.pb` model?

Excuse me @LupusSanctus, I noticed now that I' haven't mentioned it but the problem occurs even with the pre-trained model that I linked before, so if you download the model [Mask R-CNN Inception Resnet V2](http://download.tensorflow.org/models/object_detection/tf2/20200711/mask_rcnn_inception_resnet_v2_1024x1024_coco17_gpu-8.tar.gz) and reproduce the steps I wrote, you will have the same result.

**Comment by LupusSanctus** [comment]
The problem `Unsupported data type: BOOL in function 'getMatFromTensor'` is caused by a node of type ``Loop``:
![maskrcnn_loop](https://user-images.githubusercontent.com/7347318/107539025-9f50a280-6bd5-11eb-84fc-9fd59b69f414.png)

One of its inputs has ``bool`` type, which leads to this error. OpenCV doesn't support control flow and currently there is no plan for the near future to implement it.

**Comment by kbarni** [comment]
I get this error message while loading the [tiny-yolov3](https://github.com/onnx/models/blob/main/vision/object_detection_segmentation/tiny-yolov3/model/tiny-yolov3-11.onnx) network from the official ONNX model zoo.

I'm using OpenCV 4.5.5-dev
